components/usage/pkg/db/usage.go

```diff
@@ -69,22 +71,79 @@ func FindAllDraftUsage(ctx context.Context, conn *gorm.DB) ([]Usage, error) {
 	return usageRecords, nil
 }
 
-func FindUsage(ctx context.Context, conn *gorm.DB, attributionId AttributionID, from, to VarcharTime, offset int64, limit int64) ([]Usage, error) {
+type FindUsageParams struct {
+	AttributionId AttributionID
+	From, To      time.Time
+	ExcludeDrafts bool
+	Order         Order
+	Offset, Limit int64
+}
+
+func FindUsage(ctx context.Context, conn *gorm.DB, params *FindUsageParams) ([]Usage, error) {
 	var usageRecords []Usage
 	var usageRecordsBatch []Usage
 
-	result := conn.WithContext(ctx).
-		Where("attributionId = ?", attributionId).
-		Where("? <= effectiveTime AND effectiveTime < ?", from.String(), to.String()).
-		Order("effectiveTime DESC").
-		Offset(int(offset)).
-		Limit(int(limit)).
-		FindInBatches(&usageRecordsBatch, 1000, func(_ *gorm.DB, _ int) error {
-			usageRecords = append(usageRecords, usageRecordsBatch...)
-			return nil
-		})
+	db := conn.WithContext(ctx).
+		Where("attributionId = ?", params.AttributionId).
+		Where("? <= effectiveTime AND effectiveTime < ?", params.From, params.To)
+	if params.ExcludeDrafts {
+		db = db.Where("draft = ?", false)
+	}
+	db = db.Order(fmt.Sprintf("effectiveTime %s", params.Order.ToSQL()))
+	if params.Offset != 0 {
+		db = db.Offset(int(params.Offset))
+	}
+	if params.Limit != 0 {
+		db = db.Limit(int(params.Limit))
+	}
+
+	result := db.FindInBatches(&usageRecordsBatch, 1000, func(_ *gorm.DB, _ int) error {
+		usageRecords = append(usageRecords, usageRecordsBatch...)
+		return nil
+	})
 	if result.Error != nil {
 		return nil, fmt.Errorf("failed to get usage records: %s", result.Error)
 	}
 	return usageRecords, nil
 }
+
```

Comment:
For implementing these, you may [find gorm scopes very applicable](https://gorm.io/docs/scopes.html)

Reply:
Not quite sure how you would like them to be used in this context.